satpy/tests/reader_tests/test_eum_l2_bufr.py

```diff
@@ -0,0 +1,351 @@
+        _ = bufr_obj.get_data('longitude', '#1#longitude', coordinates=False)  # populate the file handler with these data
+
+        z = bufr_obj.get_data(dataset_name='TestData', key=TEST_DATA[input_file]['key'], coordinates=True)
+
+        ad = bufr_obj.fh.get_area_def(None)
+        assert ad == TEST_DATA[input_file]['area']
+        data_1d = np.concatenate((DATA, DATA), axis=0)
+
+        # Put BUFR data on 2D grid that the 2D array returned by get_dataset should correspond to
+        lons_1d, lats_1d = da.compute(bufr_obj.fh.longitude, bufr_obj.fh.latitude)
+        icol, irow = ad.get_array_indices_from_lonlat(lons_1d, lats_1d)
+
+        data_2d = np.empty(ad.shape)
+        data_2d[:] = np.nan
+        data_2d[irow.compressed(), icol.compressed()] = data_1d[~irow.mask]
+        np.testing.assert_array_equal(z.values, data_2d)
+
+        # Removed assert dedicated to products with seg_size=3 (covered by GII test case)
+
+    def test_data_with_rect_lon(self, input_file):
+        """Test data loaded with AreaDefinition and user defined rectification longitude."""
+        bufr_obj = L2BufrData(input_file, with_adef=True)
+        np.testing.assert_equal(bufr_obj.fh.ssp_lon, int(TEST_DATA[input_file]['RectificationLongitude'][1:])/10)
+        _ = bufr_obj.get_data('latitude', '#1#latitude', coordinates=False)  # We need to load the lat/lon data in order to
+        _ = bufr_obj.get_data('longitude', '#1#longitude', coordinates=False)  # populate the file handler with these data
+        _ = bufr_obj.get_data(dataset_name='TestData', key=TEST_DATA[input_file]['key'], coordinates=True)
+        # We need to lead the data in order to create the AreaDefinition
+
+        ad = bufr_obj.fh.get_area_def(None)
+        assert ad == TEST_DATA[input_file]['area']
+
+
+class AMVBufrData:
+    """Mock SEVIRI L2 AMV BUFR data."""
+
+    @unittest.skipIf(sys.platform.startswith('win'), "'eccodes' not supported on Windows")
+    def __init__(self, filename):
+        """Initialize by mocking test data for testing the SEVIRI L2 BUFR reader."""
+        from satpy.readers.eum_l2_bufr import EumetsatL2BufrFileHandler
+
+        with mock.patch('satpy.readers.eum_l2_bufr.np.fromfile'):
+            FILENAME_INFO = {'start_time': '20191112000000',
+                             'spacecraft': 'MSG3',
+                             'server': 'TESTSERVER'}
+            self.fh = EumetsatL2BufrFileHandler(filename, FILENAME_INFO,
+                                filetype_info={'file_type': 'seviri_l2_bufr_amv'},
+                                with_area_definition=True)
+
+
+class TestAMVBufrReader:
+    """Test SEVIRI L2 BUFR Reader for AMV data."""
+
+    @staticmethod
+    def test_amv_with_area_def():
+        """Test that AMV data can not be loaded with an area definition."""
+        bufr_obj = AMVBufrData('AMVBUFRProd_20231023044500Z_00_OMPEFS03_MET10_FES_E0000')
+        assert bufr_obj.fh.with_adef is False
```

Comment:
Please remove reference to SEVIRI and add test case also with FCI/L2PF AMV BUFR file.